File: Day2.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Day2 {
    public static void main(String[] args) {
        File file = new File("day2.txt");
        part1(file); part2(file);
    }

    public static void part2(File file) {
        Scanner scan = null;
        try {
            scan = new Scanner(file);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        int h = 0, d = 0, aim = 0;
        while (scan.hasNextLine()) {
            String[] line = scan.nextLine().split(" ");
            if (line[0].equals("forward")) {
                h += Integer.parseInt(line[1]);
                d += aim*Integer.parseInt(line[1]);
            } else if (line[0].equals("down")) {
                aim += Integer.parseInt(line[1]);
            } else if (line[0].equals("up")) {
                aim -= Integer.parseInt(line[1]);
            } 
        }
        System.out.println(d*h);
    }

    public static void part1(File file) {
        Scanner scan = null;
        try {
            scan = new Scanner(file);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        int h = 0, d = 0;
        while (scan.hasNextLine()) {
            String[] line = scan.nextLine().split(" ");
            if (line[0].equals("forward")) {
                h += Integer.parseInt(line[1]);
            } else if (line[0].equals("down")) {
                d += Integer.parseInt(line[1]);
            } else if (line[0].equals("up")) {
                d -= Integer.parseInt(line[1]);
            } 
        }
        System.out.println(d*h);
    }
}

File: day2.py
f = open('day2.txt').readlines()

# part one
horizonal, depth = 0, 0
for line in f:
  line = line.split(' ')
  direction = line[0]
  amount = int(line[1])
  if direction == 'forward':
    horizonal += amount
  elif direction == 'down':
    depth += amount
  else:
    depth -= amount
print(depth * horizonal)

# part two
horizonal, depth, aim = [0]*3
for line in f:
  line = line.split(' ')
  direction = line[0]
  amount = int(line[1])
  if direction == 'down':
    aim += amount
  elif direction == 'up':
    aim -= amount
  else:
    horizonal += amount
    depth += aim*amount
print(depth * horizonal)
